src/ode_solver/ode_solver_base.cpp

```diff
@@ -188,7 +191,7 @@ int ODESolverBase<dim,real,MeshType>::steady_state ()
             && (meshadaptation->current_refinement_cycle < meshadaptation->mesh_adaptation_param->total_refinement_cycles))
         {
             meshadaptation->adapt_mesh();
-            allocate_ode_system ();
+            allocate_ode_system (); // why is this being done here?
```

Comment:
@PranshulThakur , what's with the mesh adaptation and solution output stuff doing together and here?

In the future, we might want to think of a way to separate the concerns of solving an ODE and adaptive refinement such that ODE solve knows nothing about grids.

Reply:
I think it would make sense to move appropriate contents of steady_state() into flow_solver.cpp similar to how the contents of advance_solution_time() were implemented in flow_solver.cpp for unsteady flow using step_in_time(). That way operations like outputting files, writing tables, and mesh adaptation can be done on the fly in a more appropriate class; offering more freedom. I see these "operations" as tasks for a flow solver rather than an ODE solver. @donovan97

Reply:
I would tend to agree with @jbrillon here, might be good to add this as an issue.

Reply:
Yeah, some tests which perform mesh adaptation (like the S-shock) do not use flow solver yet. I will add this as an issue for now and create a separate PR to move MeshAdaptation and all tests which use it to flow solver.